src/handler/operation/cookie.js

```diff
@@ -67,9 +68,13 @@ function tabToStringUrl(tab) {
   return `${url.protocol}//${url.hostname}`;
 }
 
-function tabToStringDomain(tab) {
-  const url = new URL(tab.url);
+function tabToStringDomain(inputUrl) {
+  const url = new URL(inputUrl.startsWith("http") ? inputUrl : `http://${inputUrl}`);
   return `.${url.hostname}`;
 }
 
+function isDomainMatch(itemDomain, cookieDomain) {
+  return itemDomain === cookieDomain || "." + itemDomain === cookieDomain;
+}
```

Comment:
possible misconception from my side, but is the next viable?
```suggestion
function isDomainMatch(itemDomain, cookieDomain) {
  return itemDomain === cookieDomain.substring(1);
}
```

Reply:
From my experience, we still need to validate the `.domain.com`and `domain.com` since I've seen both cases. Although it seems that the [extra dot is not necessary](https://stackoverflow.com/a/20884869
).
But using the substring seems a neater approach. Thank you for the suggestion.